src/bindings/python/src/openvino/runtime/ie_api.py

```diff
@@ -23,30 +24,47 @@
 )
 
 
-class Model(ModelBase):
+class Model:
     def __init__(self, *args: Any, **kwargs: Any) -> None:
         if args and not kwargs:
             if isinstance(args[0], ModelBase):
-                super().__init__(args[0])
+                self.__model = ModelBase(args[0])
             elif isinstance(args[0], Node):
-                super().__init__(*args)
+                self.__model = ModelBase(*args)
             else:
-                super().__init__(*args)
+                self.__model = ModelBase(*args)
         if args and kwargs:
-            super().__init__(*args, **kwargs)
+            self.__model = ModelBase(*args, **kwargs)
         if kwargs and not args:
-            super().__init__(**kwargs)
+            self.__model = ModelBase(**kwargs)
+
+    def __getattr__(self, name: str) -> Any:
+        if hasattr(self.__model, name):
+            return getattr(self.__model, name)
```

Comment:
How does this behave when `hasattr(self.__model, name)` returns `false`? If nothing happens (there's no `else`) then we probably need to raise an error explaining what happend.

Perhaps an appropriate error would be thrown if we just delete the `if` - maybe `getattr` throws something user friendly when the attr cannot be found?

Reply:
Currently it returns None for properties and `TypeError: 'NoneType' object is not callable` for methods

Without `if hasattr` `AttributeError` is raised properly e.g.
`E       AttributeError: 'openvino._pyopenvino.Model' object has no attribute 'made_up_attr'`
but there is a change in behavior when somebody tries to access Model after it goes outside of with-expression. [Here](https://github.com/almilosz/openvino/blob/a506b59577575d20fc48387ab2b8ab794a7eb478/src/bindings/python/tests/test_runtime/test_model.py#L826)  `model` will be None. Not only it's attributes.

Reply:
Could we combine the advantages of both cases? Keep the `if hasattr`, but in `else` raise an `AttributeError` similar to the one you pasted. We'll keep the original behavior and have a much more user-friendly error message than `TypeError: 'NoneType' object is not callable`.